src/components/scenes/FioPendingRequestDetailsScene.js

```diff
@@ -0,0 +1,316 @@
+import styles from '../../styles/scenes/FioPendingRequestDetailsStyle.js'
+import type { State } from '../../types/reduxTypes'
+import type { GuiWallet } from '../../types/types'
+import { FormField } from '../common/FormField'
+import { SceneWrapper } from '../common/SceneWrapper'
+import { Airship, showError } from '../services/AirshipInstance'
+
+export type FioPendingRequestDetailsStateProps = {
+  fromWallet: GuiWallet,
+  toWallet: GuiWallet,
+
+  // The following props are used to populate the confirmation modal
+  fromCurrencyCode: string,
+  toCurrencyCode: string,
+
+  // Number of times To and From wallets were flipped
+  wallets: { [string]: GuiWallet },
+  supportedWalletTypes: Array<Object>,
+  state: State,
+  exchangeRates: ExchangeRatesState,
+
+  fioWalletByAddress: EdgeCurrencyWallet | null,
+  exchangeDenomination: EdgeDenomination,
+  selectedFioPendingRequest: any,
+  isoFiatCurrencyCode: any
+}
+
+export type FioPendingRequestDetailsDispatchProps = {
+  onSelectWallet: (walletId: string, currencyCode: string) => void,
+  openModal(data: 'from' | 'to'): mixed,
+  fioAcceptRequest: (
+    fioWalletByAddress: EdgeCurrencyWallet,
+    pendingRequest: Object,
+    payerPublicAddress: string,
+    txId: string,
+    notes?: string,
+    fee: number,
+    cb: Function
+  ) => any,
+  createCurrencyWallet: (walletType: string, currencyCode: string) => void,
+  setFioWalletByFioAddress: string => void
+}
+
+type Props = FioPendingRequestDetailsStateProps & FioPendingRequestDetailsDispatchProps
+
+type LocalState = {
+  memo: string
+}
+
+export class FioPendingRequestDetailsComponent extends Component<Props, LocalState> {
+  constructor (props: Props) {
+    super(props)
+    const newState: LocalState = {
+      memo: this.props.selectedFioPendingRequest.content.memo
+    }
+    this.state = newState
+  }
+
+  componentDidMount (): void {
+    this.props.setFioWalletByFioAddress(this.props.selectedFioPendingRequest.payer_fio_address)
+  }
+
+  fiatAmount = (currencyCode: string, amount: string) => {
+    const { exchangeRates, isoFiatCurrencyCode } = this.props
+    let fiatPerCrypto
+    if (currencyCode === Constants.FIO_STR) {
+      fiatPerCrypto = 1
+    } else {
+      const rateKey = `${currencyCode}_${isoFiatCurrencyCode}`
+      fiatPerCrypto = exchangeRates[rateKey] ? exchangeRates[rateKey] : 0
+    }
+    const amountToMultiply = parseFloat(amount)
+
+    return (fiatPerCrypto * amountToMultiply).toFixed(2)
+  }
+
+  amountField = (amount: string, currencyCode: string, cryptoSymbol: string, currencySymbol: string) => {
+    return (
+      <View style={styles.row}>
+        <T style={styles.title}>
+          Amount: {amount} {cryptoSymbol} ({currencySymbol}
+          {this.fiatAmount(currencyCode, amount)})
+        </T>
+      </View>
+    )
+  }
+
+  requestedField = (payee: string) => {
+    return (
+      <View style={styles.row}>
+        <T style={styles.title}>Request from</T>
```

Comment:
Please add to strings file for translation

Reply:
sorry, missed this, added